src/pages/tasks/common/hooks.tsx

```diff
@@ -0,0 +1,210 @@
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2022. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+import { Link } from '@invoiceninja/forms';
+import { date } from 'common/helpers';
+import { route } from 'common/helpers/route';
+import { useFormatMoney } from 'common/hooks/money/useFormatMoney';
+import { useCurrentCompany } from 'common/hooks/useCurrentCompany';
+import { useCurrentCompanyDateFormats } from 'common/hooks/useCurrentCompanyDateFormats';
+import { useCurrentUser } from 'common/hooks/useCurrentUser';
+import { Task } from 'common/interfaces/task';
+import { customField } from 'components/CustomField';
+import { DataTableColumnsExtended } from 'pages/invoices/common/hooks/useInvoiceColumns';
+import { useTranslation } from 'react-i18next';
+import { TaskStatus } from './components/TaskStatus';
+import { calculateEntityState } from './helpers/calculate-entity-state';
+import { calculateTime } from './helpers/calculate-time';
+
+export const taskColumns = [
+  'status',
+  'number',
+  'client',
+  //   'project',  @Todo: Need to fetch relationship
+  'description',
+  'duration',
+  'entity_state',
+  'archived_at',
+  //   'assigned_to', @Todo: Need to fetch relationship
+  //   'calculated_rate', @Todo: Diff vs `rate`?
```

Comment:
@turbo124 what's difference between `rate` and `calulated_rate`?

Reply:
@beganovich this appears to be a synthetic calculated property cc @hillelcoren for confirmation.

Reply:
The rate is the value set directly on the record, the calculated rate checks the task/group/company to determine the rate